src/Integration.Vsix.UnitTests/SonarLintTagger/TaggerProviderTests.cs

```diff
@@ -265,6 +265,56 @@ public void RequestAnalysis_Should_NotThrow_When_AnalysisFails()
             act.Should().NotThrow();
         }
 
+        [TestMethod]
+        [DataRow(null)]
+        [DataRow(new string[] { })]
+        public void FilterIssueTrackersByPath_NullOrEmptyPaths_AllTrackersReturned(string[] filePaths)
+        {
+            var tracker1 = CreateMockedIssueTracker("any");
+            var tracker2 = CreateMockedIssueTracker("any2");
+
+            var actual = TaggerProvider.FilterIssuesTrackersByPath(new IIssueTracker[] { tracker1, tracker2}, filePaths);
+            actual.Should().BeEquivalentTo(tracker1, tracker2);
+        }
+
+        [TestMethod]
+        public void FilterIssueTrackersByPath_WithPaths_ExpectedTrackersReturned()
```

Comment:
I tried using DataRows for this test too but it got too ugly and difficult to read.